public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -292,6 +293,23 @@ export class PanelModel implements DataConfigSource {
     this.gridPos.h = newPos.h;
   }
 
+  runAllPanelQueries(dashboardId: number, dashboardTimezone: string, timeData: TimeOverrideResult, width: number) {
+    this.getQueryRunner().run({
+      datasource: this.datasource,
+      queries: this.targets,
+      panelId: this.editSourceId || this.id,
+      dashboardId: dashboardId,
+      timezone: dashboardTimezone,
+      timeRange: timeData.timeRange,
+      timeInfo: timeData.timeInfo,
+      maxDataPoints: this.maxDataPoints || width,
+      minInterval: this.interval,
+      scopedVars: this.scopedVars,
+      cacheTimeout: this.cacheTimeout,
+      transformations: this.transformations,
+    });
+  }
+
```

Comment:
Hmm, I'm not 100% sure this is a good idea. Now we'll have both `getQueryRunner` and `runAllPanelQueries` on PanelModel that can be used to do the same.

Reply:
I agree with Hugo about this, think we don't need this abstraction.

Reply:
🤔 you have a good point, but I am not sure, the idea of creating this abstraction was to be able to reuse it on `PanelChrome` and `PanelEditorTableView` components without passing all the panel properties that the `run` function needs.

If I remove the abstraction, it means we will use directly the `getQueryRunner().run`  in those components and we need to pass all the panel props, is it the approach we want to follow? @hugohaggmark @dprokop @torkelo

Reply:
Aha, I see what you mean know @axelavargas but aren't the properties needed already on `panel` & `dashboard`? But I'm not on the fences on this one so you can leave it like this.